payment/controllers.go

```diff
@@ -595,6 +621,47 @@ func MerchantTransactions(service *Service) handlers.AppHandler {
 	})
 }
 
+// HandleStripeWebhook is the handler for stripe checkout session webhooks
+func HandleStripeWebhook(service *Service) handlers.AppHandler {
+	return handlers.AppHandler(func(w http.ResponseWriter, r *http.Request) *handlers.AppError {
+		stripe.Key = os.Getenv("STRIPE_SECRET")
+		endpointSecret := os.Getenv("STRIPE_WEBHOOK_SECRET")
+
+		const MaxBodyBytes = int64(65536)
+		r.Body = http.MaxBytesReader(w, r.Body, MaxBodyBytes)
+
+		body, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			return handlers.WrapError(err, "error reading request body", http.StatusServiceUnavailable)
+		}
+
+		event, err := webhook.ConstructEvent(body, r.Header.Get("Stripe-Signature"), endpointSecret)
+		if err != nil {
+			return handlers.WrapError(err, "error verifying webhook signature", http.StatusBadRequest)
+		}
+
+		// Fulfill order only if checkout session completed
+		if event.Type == "checkout.session.completed" {
+			var session stripe.CheckoutSession
+			err := json.Unmarshal(event.Data.Raw, &session)
+			if err != nil {
+				return handlers.WrapError(err, "error parsing webhook JSON", http.StatusBadRequest)
+			}
+			orderID, err := uuid.FromString(session.ClientReferenceID)
+			if err != nil {
+				return handlers.WrapError(err, "error retrieving client reference ID", http.StatusBadRequest)
+			}
+			err = service.Datastore.UpdateOrder(orderID, "paid")
+			if err != nil {
+				return handlers.WrapError(err, "error updating order status", http.StatusInternalServerError)
+			}
+			return handlers.RenderContent(r.Context(), "checkout successful", w, http.StatusOK)
+		}
+
+		return handlers.RenderContent(r.Context(), "error in checkout session", w, http.StatusBadRequest)
```

Comment:
@dlipeles : stripe will repeat the webhook call in non OK response case... if the event.Type is not "checkout.session.completed" do we want stripe to attempt the webhook again and again for days?  I think I would suggest for the event.Type's that we dont support we just pass back an OK status.